static/js/admin.js

```diff
@@ -42,6 +42,13 @@ var admin_settings_label = {
 };
 
 exports.setup_page = function () {
+    var can_add_emojis = false;
+    if (page_params.is_admin) {
+        can_add_emojis = true;
+    } else if (!(page_params.realm_add_emoji_by_admins_only || page_params.is_guest)) {
+        can_add_emojis = true;
```

Comment:
I know this is only 5 lines of code, but I think it's worth extracting to a function, which will allow you to flatten the code a bit and add direct tests to it:

~~~ js
exports.can_add_emoji = function () {
    if (page_params.is_guest) {
        return false;
    }

    if (page_params.is_admin) {
        return true;
    }

    // for normal users, we depend on the setting
    return !page_params.realm_add_emoji_by_admins_only;
};
~~~

Does that make sense?

Reply:
Yeah, makes sense.  Most of the existing code is probably fine to leave as is, but if you see any opportunities for quick cleanup, it's nice to be able to test all these permission things directly.